### DISCUSSION TITLE
Large Page Data warnings for SSR-rendered blog content

### DISCUSSION THREAD

**Discussion by dmitrizzle** [vercel/next.js#60193] [Help]
### Summary

Many of the pages on my blog get this error in the Node console: `Warning: data for page ... is XXX kB which exceeds the threshold of 128 kB. This amount of data can reduce performance.`

I don't think there's a way to reduce this amount of data (explained below). However, **all initial requests are cached on CDN.**

Is bypassing this warning by modifying `largePageDataBytes` the only solution for my scenario?
What do you think would be the major drawback of ignoring this error?

***

I don't think there's a way to reduce the amount of data because most of it is static content. The app uses that content to render text formatting and the text itself for blog articles that can sometimes be over 10,000 words long. This content has to be rendered on the server via SSR for accessibility and SEO. Some parts of that content are interactive (i.e. clicking an image brings up a modal), but the code for those interactions is not part of the data. Overall, I'm pretty good at loading content dynamically if it doesn't have to appear via SSR.

Thanks!

**Comment by icyJoseph** [comment]
Many projects end up crossing that threshold. The number, 128kb, is itself, only a good estimate by the Next.js team.

You might want to make sure that you are not sending unnecessary data though. Hydration does require the same inputs used in the server, to replay the client component tree. 

Say, you have a list of elements, and you use that as props, either from a Server Component to a Client Component, or with data fetching functions (gSP, gSSP) on pages router. Then you filter that list as part of the render pass. In this case, you could've filtered before you passed it to the client component, or before returning the data-fetching function, to reduce the amount of data sent to the client.

It is not always possible, but it is worth taking a look at what data is being transferred.

Here's a helpful video as well: https://www.youtube.com/watch?v=FfHsIio4aCU
